### PR TITLE
Add ChromBackendMzR backend and additional variables

### PR DESCRIPTION
**Pull request by jorainer**
- Add `ChromBackendMzR` backend (issue #7).
- Add additional chromatom variables `chromIndex` and `collisionEnergy`.

### CODE REVIEW

File: R/ChromBackendMzR-functions.R [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    hdr$productMzMin <- hdr$productMz - hdr$productMzMin
+    hdr$productMzMin <- hdr$productMz + hdr$productMzMax
+    DataFrame(hdr)
+}
+
+#' Read chromatogram data from a single mzML file.
+#'
+#' @param x `character(1)` with the file to read from.
+#'
+#' @param chromIndex (required) indices of chromatograms from which the data
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/ChromBackendMzR-functions.R]: Well, yes and no. The *chromatogram variable* providing this information is called `chromIndex` that's why I used here `chromIndex` too. All chromatogram variables are singular, such as `rtime`, `msLevel` (not `rtimes`, `msLevels` etc). In addition I wanted to be consistent with `Spectra`, where the corresponding variable is called `scanIndex`.

File: R/ChromBackendMzR-functions.R [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+#' @noRd
+.mzR_chromatograms <- function(x = character(), chromIndex = integer()) {
+    if (length(x) != 1)
+        stop("'x' should have length 1")
+    msd <- mzR::openMSfile(x)
+    on.exit(mzR::close(msd))
+    chrs <- mzR::chromatogram(msd, chromIndex)
+    if (is.data.frame(chrs))
+        chrs <- list(chrs)
+    lapply(chrs, function(z) {
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/ChromBackendMzR-functions.R]: Nice! Thanks for the suggestion!

File: R/ChromBackend.R [hunk 3]
```diff
@@ -387,13 +417,39 @@ setReplaceMethod("chromNames", "ChromBackend", function(object, value) {
     stop("Not implemented for ", class(object), ".")
 })
 
+#' @exportMethod chromIndex
+#'
+#' @rdname ChromBackend
+setMethod("chromIndex", "ChromBackend",
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/ChromBackend.R]: As discussed above, I would keep it in singular. Same as `rtime`, `msLevel` etc.

File: R/ChromBackendMzR.R [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    stop(class(object), " does not support replacing m/z values")
+})
+
+#' @rdname hidden_aliases
+setReplaceMethod("$", "ChromBackendMzR", function(x, name, value) {
+    if (name == "rtime" || name == "intensity")
+        stop("'ChromBackendMzR' does not support replacing retention time ",
+             "or intensity values")
+    if (length(value) == 1)
+        value <- rep(value, length(x))
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/ChromBackendMzR.R]: Good point. I will change (here and in `Spectra`).

[the remark this answers is not in the document]
Comment (jorainer) [reply R/ChromBackendMzR.R]: Done in https://github.com/rformassspectrometry/Chromatograms/pull/9/commits/9febc982c8c5eac86f51d05ddcbd931e994c9c87

File: R/ChromBackendMzR-functions.R [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+#'
+#' @noRd
+.mzR_chrom_header <- function(x = character()) {
+    if (length(x) != 1)
+        stop("'x' should have length 1")
+    requireNamespace("mzR", quietly = TRUE)
+    msd <- mzR::openMSfile(x)
+    on.exit(mzR::close(msd))
+    hdr <- mzR::chromatogramHeader(msd)
+    colnames(hdr)[colnames(hdr) == "chromatogramIndex"] <- "chromIndex"
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/ChromBackendMzR-functions.R]: Sure. Makes sense. Will change.

[the remark this answers is not in the document]
Comment (jorainer) [reply R/ChromBackendMzR-functions.R]: Done in https://github.com/rformassspectrometry/Chromatograms/pull/9/commits/12797febba7997b914b5621de7b9fee9dc02d073

File: R/ChromBackend.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (sgibb) [reply R/ChromBackend.R]: fine

**Pull request merged**